controllers/handlers/passt/passt.go

```diff
@@ -0,0 +1,345 @@
+package passt
+
+import (
+	"fmt"
+	"maps"
+	"os"
+	"sync"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	securityv1 "github.com/openshift/api/security/v1"
+
+	kubevirtcorev1 "kubevirt.io/api/core/v1"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/operands"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/nodeinfo"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	DeployPasstNetworkBindingAnnotation = "deployPasstNetworkBinding"
+
+	BindingName = "passt"
+
+	networkBindingNADName       = "primary-udn-kubevirt-binding"
+	networkBindingNADNamespace  = "default"
+	NetworkAttachmentDefinition = networkBindingNADNamespace + "/" + networkBindingNADName
+
+	bindingComputeMemoryOverhead = "250Mi"
+)
+
+var (
+	passtResourceMemory = resource.MustParse(bindingComputeMemoryOverhead)
+	passtImage          string
+	passtImageOnce      sync.Once
+)
+
+// CheckPasstImagesEnvExists checks if the passt image environment variable exists
+func CheckPasstImagesEnvExists() error {
+	if _, passtImageVarExists := os.LookupEnv(hcoutil.PasstImageEnvV); !passtImageVarExists {
+		return fmt.Errorf("the %s environment variable must be set", hcoutil.PasstImageEnvV)
+	}
+	if _, passtCNIImageVarExists := os.LookupEnv(hcoutil.PasstCNIImageEnvV); !passtCNIImageVarExists {
+		return fmt.Errorf("the %s environment variable must be set", hcoutil.PasstCNIImageEnvV)
+	}
+	return nil
+}
+
+// NetworkBinding creates an InterfaceBindingPlugin for passt network binding
+func NetworkBinding(sidecarImage string) kubevirtcorev1.InterfaceBindingPlugin {
```

Comment:
Why do we need to pass the image as parameter? the function only called once. The same is for the GetImage function - it is only called once, to pass the parameter to this function. We can just call GetImage from within the function.

In addition, the content of the function is actually static. It can be a variable instead of a function. May the function can just return the var.

Reply:
Called `GetImage` directly from `NetworkBinding` (will push soon), unless you meant more things

not sure i understood the last paragraph, do you suggest to just inline the body of NetworkBinding where it is called ? i think it is nicer to have it in passt file, instead inline on kubevirt file, but whatever you choose

fwiw NetworkBinding is called only at the kubevirt handler (via cache), 
but more times indirectly via `NewKubeVirt` (on `pkg/webhooks/validator/validator.go`)